src/cudadecoder/batched-static-nnet3-kernels.cu

```diff
@@ -0,0 +1,167 @@
+					iframe += gridDim.y) {
+				for (int idim = threadIdx.x; idim < params.input_dim; idim += blockDim.x) {
+					if(iframe < n_left_context_frames_from_frame0) {
+						d_batch_slot_with_context[iframe * params.d_batch_with_context_frame_stride+ idim] = d_batch_slot_features[0 + idim]; // frame 0
+					}
+					else if (iframe < (n_left_context_frames_from_frame0+batch_assign.n_frames_already_in_context)) {
+						// Those are the frames coming from context
+						int src_iframe_in_saved_context = iframe-n_left_context_frames_from_frame0;
+						d_batch_slot_with_context[iframe * params.d_batch_with_context_frame_stride+ idim] = d_channel_context[src_iframe_in_saved_context*params.d_all_context_frames_frame_stride+ idim];
+					} else {
+						// Now we are moving the frames coming from the new chunk
+						int src_iframe_in_new_chunk = iframe-n_left_context_frames_from_frame0-batch_assign.n_frames_already_in_context;
+						d_batch_slot_with_context[iframe * params.d_batch_with_context_frame_stride+ idim] = d_batch_slot_features[src_iframe_in_new_chunk*params.d_features_frame_stride + idim];
+					}
+				}
+
+				if(iframe == 0 && params.d_batch_ivectors) { // one CTA moves the ivectors
+					for (int idim = threadIdx.x; idim < params.ivector_dim; idim += blockDim.x) {
+						params.d_batch_ivectors[batch_slot*params.d_batch_ivectors_stride +idim] = batch_assign.d_ivectors[idim];
+					}
+				}
+			}
+		} 
+	}
+
+void BuildBatchWithContextKernel(const dim3 &grid, const dim3 &block,
+                                 const cudaStream_t &stream,
+                                 const BatchedStaticNnet3KernelParams &params) {
+  build_batch_with_context_kernel<<<grid, block, 0, stream>>>(params);
+}
+
+	__global__ void build_batch_with_context_context_flush_kernel(
+			BatchedStaticNnet3KernelParams params) {
+		for (int batch_slot = blockIdx.z; batch_slot < params.batch_size;
+				batch_slot += gridDim.z) {
+
+			BatchSlotAssignment batch_assign =
+				params.d_batch_slot_assignement[batch_slot];
+			BaseFloat *d_channel_context = &params.d_all_context_frames[batch_assign.ichannel *
+				params.d_all_context_frames_channel_stride];
+			BaseFloat *d_batch_slot_with_context = &params.d_batch_with_context[params.d_batch_with_context_batch_stride * batch_slot];
+
+
+			int n_frames_in_context =
+				batch_assign.n_frames_already_in_context;
+			int n_frames_to_set = n_frames_in_context + params.total_nnet_right_context;  
+//			printf("will set %i frames, %i in context \n", n_frames_to_set, n_frames_in_context);
+
+			for (int iframe = blockIdx.y; iframe < n_frames_to_set;
+					iframe += gridDim.y) {
+				for (int idim = threadIdx.x; idim < params.input_dim; idim += blockDim.x) {
+					if(iframe < n_frames_in_context) {
+float f=						d_batch_slot_with_context[iframe * params.d_batch_with_context_frame_stride+ idim] = d_channel_context[iframe*params.d_all_context_frames_frame_stride + idim];
```

Comment:
very strange whitespace-- perhaps your tab settings were wrong?  Please make sure to convert all tabs to spaces, as google style guide mandates.  And left-justified `float f=` is wrong, plus shoul dbe space after =.

Reply:
My guess is my new clang-format setup doesn't work properly with .cu files. I'll fix those

Reply:
It was using llvm style for cuda files. Back to google style now.